src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1094,17 +1102,30 @@ func (c controller) sendCompleteInstallation(ctx context.Context, isSuccess bool
 	c.log.Infof("Done complete installation step")
 }
 
-// logClusterOperatorsStatus logging cluster operators status
-func (c controller) logClusterOperatorsStatus() {
+//Gather information about cluster operators. This information
+//is required for debugging purposes so it is printed into the
+//controller's log file (always) and sent to the service to be
+//formatted in an event to the user (in case of a failure)
+func (c controller) operatorReport() []models.OperatorMonitorReport {
 	operators, err := c.kc.ListClusterOperators()
 	if err != nil {
 		c.log.WithError(err).Warning("Failed to list cluster operators")
-		return
+		return nil
 	}
 
+	report := make([]models.OperatorMonitorReport, 0)
 	for _, operator := range operators.Items {
+		//Note: always keep this log so we have all the information
+		//in the controller's log
 		c.log.Infof("Operator %s, statuses: %v", operator.Name, operator.Status.Conditions)
+		status, message, _ := utils.MonitoredOperatorStatus(operator.Status.Conditions)
```

Comment:
I just noticed this only returns a single condition. And it would only return the first of `Available`/`Failed`, depending on the order they appear in the conditions list... This is very odd behavior. What's the reasoning behind it?